### PR TITLE
Abbreviation component

### PR DESCRIPTION
**Pull request by jubalm**
- [x] find instances on where this component can be used
- [x] create or replace test

### CODE REVIEW

**Comment by KillariDev** [comment]
is there still use for `bigintToDecimalString` or should it also be removed? The tests should also apply to this and not the old funtction

**Comment by jubalm** [comment]
I've updated the test and just realized Docker build is using node 18 which doesn't support the new Intl.NumberFormat options (roundingMode). We can either update the test, update Node to 20 (LTS) or just update the code. would like to know your opinions before i go at it again @KillariDev @MicahZoltu

File: app/ts/components/subcomponents/coins.tsx [hunk 1]
```diff
@@ -203,7 +205,9 @@ export function TokenAmount(param: TokenAmountParams) {
 	}
 	return <>
 		<CopyToClipboard content = { bigintToDecimalString(abs(param.amount), param.tokenEntry.decimals) } copyMessage = 'Token amount copied!' >
-			<p class = 'noselect nopointer' style = { style }>{ `${ sign }${ bigintToRoundedPrettyDecimalString(abs(param.amount), param.tokenEntry.decimals ) }` }&nbsp; </p>
+			<p class = 'noselect nopointer' style = { style }>
+				<AbbreviatedValue amount={param.amount} decimals={param.tokenEntry.decimals} />
```
Comment (KillariDev) [inline app/ts/components/subcomponents/coins.tsx:209]:
```suggestion
				<AbbreviatedValue amount = { param.amount } decimals = { param.tokenEntry.decimals } />
```

File: app/ts/utils/bigint.ts [hunk 2]
```diff
@@ -17,41 +17,22 @@ export function bigintToDecimalString(value: bigint, power: bigint): string {
...
-			if ( stringifiedNumber.charAt(i) !== '-' && stringifiedNumber.charAt(i) !== '.') {
-				if ( stringifiedNumber.charAt(i) !== '0') firstDigitFound = true
-				if (firstDigitFound) p++
-			}
-			if ( stringifiedNumber.charAt(i) === '.') pointFound = true
-		}
-		return roundedString.length <= significantNumbers + 1n ? parseFloat(roundedString).toString() : roundedString
-	}
+export const bigintToRoundedPrettyDecimalString = (amount: bigint, decimals?: bigint, maximumSignificantDigits = 4) => {
+	// roundingMode is not supported until https://github.com/microsoft/TypeScript/pull/56902 (merged)
```
Comment (KillariDev) [inline app/ts/utils/bigint.ts:21]:
```suggestion
```

File: app/ts/utils/bigint.ts [hunk 3]
```diff
@@ -17,41 +17,22 @@ export function bigintToDecimalString(value: bigint, power: bigint): string {
...
+	const formatOptions = { maximumSignificantDigits, notation: 'compact' } as const
+	const formatter = new Intl.NumberFormat('en-US', formatOptions )
```
Comment (KillariDev) [inline app/ts/utils/bigint.ts:23]:
```suggestion
	const formatter = new Intl.NumberFormat('en-US', formatOptions)
```

File: test/tests/utils.bigint.ts [hunk 4]
```diff
@@ -4,13 +4,13 @@ import * as assert from 'assert'
...
-		should('display 0.2 ETH', () => assert.equal(bigintToRoundedPrettyDecimalString(2n * 10n ** 17n, 18n, 4n), '0.2'))
-		should('display 00000001 ETH', () => assert.equal(bigintToRoundedPrettyDecimalString(10n ** 10n, 18n, 4n), '0.00000001'))
-		should('display 100M ETH', () => assert.equal(bigintToRoundedPrettyDecimalString(10n ** 26n, 18n, 4n), '100M'))
-		should('display 2.345k ETH', () => assert.equal(bigintToRoundedPrettyDecimalString(234567n * 10n ** 16n, 18n, 4n), '2.345k'))
-		should('display -2.345k ETH', () => assert.equal(bigintToRoundedPrettyDecimalString(-234567n * 10n ** 16n, 18n, 4n), '-2.345k'))
-		should('display -0.2345 ETH', () => assert.equal(bigintToRoundedPrettyDecimalString(-234567n * 10n ** 12n, 18n, 4n), '-0.2345'))
+		should('display 1 ETH', () => assert.equal(bigintToRoundedPrettyDecimalString(10n ** 18n, 18n, 4), '1'))
+		should('display 0.2 ETH', () => assert.equal(bigintToRoundedPrettyDecimalString(2n * 10n ** 17n, 18n, 4), '0.2'))
+		should('display 00000001 ETH', () => assert.equal(bigintToRoundedPrettyDecimalString(10n ** 10n, 18n, 4), '0.00000001'))
```
Comment (KillariDev) [inline test/tests/utils.bigint.ts:9]:
```suggestion
		should('display 0.00000001 ETH', () => assert.equal(bigintToRoundedPrettyDecimalString(10n ** 10n, 18n, 4), '0.00000001'))
```

**Review by KillariDev — CHANGES_REQUESTED** [review]
when performing a swap, this is what I see:
image

**Comment by jubalm** [comment]
> when performing a swap, this is what I see: image

just noticed there was a missing commit, patched on [76ba86f](https://github.com/DarkFlorist/TheInterceptor/pull/949/commits/76ba86f2a04b75e228bd06ecc8c66dfd150314d0) as well as the comments

File: app/ts/components/subcomponents/AbbreviatedValue.tsx [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import { ComponentChild } from 'preact'
+import { bigintToNumberFormatParts } from '../../utils/bigint.js'
+
+export const AbbreviatedValue = ({ amount, decimals = 18n }: { amount: bigint, decimals?: bigint }) => {
+	const numberParts = bigintToNumberFormatParts(amount, decimals)
+	const domElement: ComponentChild[] = []
+
+	for (const { type, value } of numberParts) {
+		if (type === 'fraction') {
+			const significantDigits = `${Number(value)}`
```
Comment (KillariDev) [inline app/ts/components/subcomponents/AbbreviatedValue.tsx:10]:
```suggestion
			const significantDigits = `${ Number(value) }`
```

**Comment by jubalm** [comment]
Added additional tests that account for prefixed values 10000 and beyond (and negative equivalent)

**Pull request merged**